src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,35 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save country to DB : " + country.getName()
+            );
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Country country = session.get(Country.class, id);
+            session.close();
+            return Optional.ofNullable(country);
+        } catch (RuntimeException e) {
+            throw new DataProcessingException("Can't get a county with id : " + id);
```

Comment:
The `DataProcessingException` constructor call in the catch block of the `get` method is missing the cause exception `e`. Consider adding it to provide more context for the exception: `throw new DataProcessingException("Can't get a country with id : " + id, e);`